vllm/v1/engine/core.py

```diff
@@ -247,25 +249,37 @@ def __init__(
         executor_class: Type[Executor],
         log_stats: bool,
     ):
-        super().__init__(vllm_config, executor_class, log_stats)
-
-        # Background Threads and Queues for IO. These enable us to
-        # overlap ZMQ socket IO with GPU since they release the GIL,
-        # and to overlap some serialization/deserialization with the
-        # model forward pass.
-        # Threads handle Socket <-> Queues and core_busy_loop uses Queue.
-        self.input_queue: queue.Queue[Tuple[EngineCoreRequestType,
-                                            Any]] = queue.Queue()
-        self.output_queue: queue.Queue[EngineCoreOutputs] = queue.Queue()
-        threading.Thread(target=self.process_input_socket,
-                         args=(input_path, ),
-                         daemon=True).start()
-        threading.Thread(target=self.process_output_socket,
-                         args=(output_path, ),
-                         daemon=True).start()
-
-        # Send Readiness signal to EngineClient.
-        ready_pipe.send({"status": "READY"})
+        try:
+            super().__init__(vllm_config, executor_class, log_stats)
+
+            # Background Threads and Queues for IO. These enable us to
+            # overlap ZMQ socket IO with GPU since they release the GIL,
+            # and to overlap some serialization/deserialization with the
+            # model forward pass.
+            # Threads handle Socket <-> Queues and core_busy_loop uses Queue.
+            self.input_queue: queue.Queue[Tuple[EngineCoreRequestType,
+                                                Any]] = queue.Queue()
+            self.output_queue: queue.Queue[Union[EngineCoreOutputs,
+                                                 bytes]] = queue.Queue()
+            self.errored_sent_event = threading.Event()
+            threading.Thread(target=self.process_input_socket,
+                             args=(input_path, ),
+                             daemon=True).start()
+            threading.Thread(target=self.process_output_socket,
+                             args=(output_path, ),
+                             daemon=True).start()
+
+            # Send Readiness signal to EngineClient.
+            ready_pipe.send({"status": "READY"})
+
+        except Exception as e:
+            logger.exception("EngineCore got an Exception during startup:",
+                             exc_info=e)
+            ready_pipe.send({"status": "FAILED"})
+            raise e
+
+        finally:
+            ready_pipe.close()
```

Comment:
I'm not sure that this should go here. Probably better to catch/handle in the calling method, including handling sending the pipe ready message since it "owns" the pipe. In the failure case, I'm not sure that we need to send failed status back since the process could die for other reasons without us being able to do so, and we need to handle that equivalently from the client side.